### PR TITLE
Add contextvars from CPython 3.8

### PR DESCRIPTION
**Pull request by fanninpm**
This is currently scaffolding for the `contextvars` module introduced in CPython 3.7 (see PEP 567).

I'm welcome to any suggestions on how to flesh this out.

### CODE REVIEW

File: vm/src/stdlib/contextvars.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        const TP_FLAGS: crate::slots::PyTpFlags = crate::slots::PyTpFlags::DEFAULT;
+
+        fn impl_extend_class(_ctx: &PyContext, _class: &PyTypeRef) {
+            // TODO: RUSTPYTHON
+        }
+
+        fn extend_slots(_slots: &mut PyTypeSlots) {
+            // TODO: RUSTPYTHON
+        }
+    }
```
Comment (youknowone) [inline vm/src/stdlib/contextvars.rs:25]: are you going to write these part manually instead of writing an `#[pyimpl]`?

**Comment by fanninpm** [comment]
CPython references:

- https://github.com/python/cpython/blob/3.8/Include/context.h
- https://github.com/python/cpython/blob/3.8/Python/context.c
- https://github.com/python/cpython/blob/3.8/Python/clinic/context.c.h
- https://github.com/python/cpython/blob/3.8/Modules/_contextvarsmodule.c
- https://github.com/python/cpython/blob/3.8/Modules/clinic/_contextvarsmodule.c.h
- https://github.com/python/cpython/blob/3.8/Include/internal/pycore_context.h

**Comment by DimitrisJim** [comment]
Bump, to you want to open an issue for the discussion on this? I think it would be better than having the discussion on the PR.

**Review by youknowone — APPROVED** [review]
I wasnt aware this was going. Looks good. Other people can work on here later.

**Pull request merged**